utils/rtty/files/rtty.init

```diff
@@ -0,0 +1,56 @@
+#!/bin/sh /etc/rc.common
+
+USE_PROCD=1
+START=99
+
+BIN=/usr/sbin/rtty
+
+start_rtty() {
+    local cfg="$1"
+    local interface ifname id description host port ssl
+
+    uci_validate_section rtty rtty "${1}" \
+        'interface:uci("network", "@interface"):br-lan' \
+        'id:maxlength(63)' \
+        'description:maxlength(126)' \
+        'host:host' \
+        'port:port' \
+        'ssl:bool:0'
+    
+    [ $? -eq 1 ] && {
+        echo "validation failed" >&2
+        return 1
+    }
+
+    [ -n "$interface" ] && ifname=$(uci -q -p /var/state get network.$interface.ifname)
+
+    [ -z "$ifname" -a -z "$id" ] && {
+        echo "You must specify an interface or ID" >&2
+        return 1
+    }
+
+    [ -z "$host" ] && {
+        echo "host required" >&2
+        return 1
+    }
+    
+    [ -z "$port" ] && {
+        echo "port required" >&2
+        return 1
+    }
```

Comment:
Maybe we can also add `minlength(1)` for these two fields.  But the current form is also good for being explicit about the error message.

Reply:
I used port:and(port,minlength(1)).
But no effect.

Reply:
I see.  It just unset the undefined options.

`
root@OpenWrt:/# validate_data system typ nam 'port:and(port,minlength(1))'
system.nam.port is unset and defaults to and(port,minlength(1)) (null)
unset -v port;
`

Reply:
No need change?